src/transactions/TransactionFrame.cpp

```diff
@@ -388,6 +413,42 @@ TransactionFrame::checkAllSignaturesUsed()
     return true;
 }
 
+void
+TransactionFrame::removeUsedOneTimeSignerKeys(LedgerDelta& delta, LedgerManager& ledgerManager)
```

Comment:
This doesn't do the right thing if a tx involves multiple accounts. For example A is the source account on the tx, but the tx has an operation where B sends to C. account B has the hash of the tx as a signer.  This won't remove the signer from B. I think line 156 above needs to also record the account the one time signer is for